src/core/DataSourceBase.ts

```diff
@@ -57,15 +80,35 @@ export abstract class DataSourceBase<TQuery extends DataQuery, TOptions extends
    * @param options - Optional configuration for the request. This can include:
    *   - `showingErrorAlert` (boolean): If true, displays an error alert on request failure.
    *   - Any other properties supported by {@link BackendSrvRequest}, such as headers, credentials, etc.
+   * @param useApiIngress - If true, uses API ingress bypassing the UI ingress for the request.
    * @returns A promise resolving to the response of type `T`.
    */
-  post<T>(url: string, body: Record<string, any>, options: Partial<BackendSrvRequest> = {}) {
+  public async post<T>(
+    url: string,
+    body: Record<string, any>,
+    options: Partial<BackendSrvRequest> = {},
+    useApiIngress = false
+  ) {
+    if (useApiIngress) {
+      const apiSession = await this.apiSessionUtils.createApiSession();
+      if (apiSession) {
```

Comment:
If API session creation fails, we should at least log an error. Possibly throw an exception.